xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -1198,30 +1198,36 @@ namespace PVR
     return false;
   }
 
-  bool CPVRGUIActions::ContinueLastPlayedChannel() const
+  bool CPVRGUIActions::PlayChannelOnStartup() const
   {
-    const CFileItemPtr item(CServiceBroker::GetPVRManager().ChannelGroups()->GetLastPlayedChannel());
-    const CPVRChannelPtr channel(item ? item->GetPVRChannelInfoTag() : CPVRChannelPtr());
-    bool bWasPlaying = false;
-    if (channel)
-    {
-      // Obtain previous 'was playing on last app quit' flag and reset it, then.
-      channel->SetWasPlayingOnLastQuit(false, bWasPlaying);
```

Comment:
As we don't use the feature at the moment we should remove the calls to `SaveLastPlayedChannel` in `CPVRManager::Deinit()` and `CPVRManager::OnSleep()`. I'll comment out the lines in a separate commit.

BTW if we re-add the feature we should reset each channel flagged as `played on last app quit` not only one. That way it never comes to an illegal state you described.